src/parser/ast/ObjectExpressionNode.h

```diff
@@ -41,9 +41,104 @@ class ObjectExpressionNode : public ExpressionNode {
         return m_properties;
     }
 
+    size_t generateInitValueByteCode(ByteCodeBlock* codeBlock, ByteCodeGenerateContext* context, PropertyNode* p, ByteCodeRegisterIndex dstRegister)
+    {
+        size_t valueIndex = p->value()->getRegister(codeBlock, context);
+        const ClassContextInformation classInfoBefore = context->m_classInfo;
+        context->m_classInfo.m_prototypeIndex = dstRegister;
+        context->m_classInfo.m_constructorIndex = SIZE_MAX;
+        context->m_classInfo.m_superIndex = SIZE_MAX;
+        p->value()->generateExpressionByteCode(codeBlock, context, valueIndex);
+        context->m_classInfo = classInfoBefore;
+        return valueIndex;
+    }
+
     virtual ASTNodeType type() override { return ASTNodeType::ObjectExpression; }
     virtual void generateExpressionByteCode(ByteCodeBlock* codeBlock, ByteCodeGenerateContext* context, ByteCodeRegisterIndex dstRegister) override
     {
+        if (m_properties.size() >= ESCARGOT_OBJECT_STRUCTURE_TRANSITION_MODE_MAX_SIZE) {
+            size_t objectCreationDataIndex = SIZE_MAX;
+            size_t initCodePosition = codeBlock->currentCodeSize();
+            if (m_properties.size()) {
```

Comment:
This if statement could be removed since `m_properties` has always elements in this case

Reply:
Oops! this if statement was needed for testing.
I am going to remove it